### ISSUE TITLE
Lua Error when Picking Up Quest while In Combat

### ISSUE THREAD

**Issue by Gogo1951** [Questie/Questie#6499]
### Description

Went to pick up a quest in combat, and got this error.

3x ...eQuestie/Modules/Quest/QuestEventHandler.lua:235: attempt to index field '?' (a nil value)
[string "@Questie/Modules/Quest/QuestEventHandler.lua"]:235: in function `HandleQuestAccepted'
[string "@Questie/Modules/Quest/QuestEventHandler.lua"]:213: in function `func'
[string "@Questie/Modules/Libs/QuestieCombatQueue.lua"]:25: in function <...eQuestie/Modules/Libs/QuestieCombatQueue.lua:17>

Locals:
self = <table> {
}
questId = 3523
cacheMiss = false
_ = <table> {
}
(*temporary) = nil
(*temporary) = "QUEST_ACCEPTED"
(*temporary) = nil
(*temporary) = "Objectives are correct. Calling accept logic. quest:"
(*temporary) = 3523
(*temporary) = "attempt to index field '?' (a nil value)"
QuestLogCache = <table> {
 questLog_DO_NOT_MODIFY = <table> {
 }
 private = <table> {
 }
}
_QuestLogUpdateQueue = <table> {
}
_QuestEventHandler = <table> {
}
questLog = <table> {
 1302 = <table> {
 }
 625 = <table> {
 }
 717 = <table> {
 }
 659 = <table> {
 }
 2988 = <table> {
 }
 543 = <table> {
 }
 762 = <table> {
 }
 2877 = <table> {
 }
 3636 = <table> {
 }
 6612 = <table> {
 }
 1449 = <table> {
 }
 2880 = <table> {
 }
 1264 = <table> {
 }
 1179 = <table> {
 }
}
QUEST_LOG_STATES = <table> {
 QUEST_ABANDONED = "QUEST_ABANDONED"
 QUEST_TURNED_IN = "QUEST_TURNED_IN"
 QUEST_REMOVED = "QUEST_REMOVED"
 QUEST_ACCEPTED = "QUEST_ACCEPTED"
}
QuestieQuest = <table> {
 private = <table> {
 }
}
QuestieJourney = <table> {
 zones = <table> {
 }
 continents = <table> {
 }
 tabGroup = <table> {
 }
 zoneMap = <table> {
 }
 private = <table> {
 }
 questCategoryKeys = <table> {
 }
}
QuestieAnnounce = <table> {
 private = <table> {
 }
}
IsleOfQuelDanas = <table> {
 localizedPhaseNames = <table> {
 }
 private = <table> {
 }
 quests = <table> {
 }
 MAX_ISLE_OF_QUEL_DANAS_PHASES = 9
}

### Questie version

10.6.0

### Game flavor

Era/HC

**Comment by BreakBB** [comment]
Did this happen right after login? 🤔 Or do you remember if you dropped combat and then immediately returned to combat?

Questie does have a check to run certain things only after combat and this is one of the cases. But somehow the code ran in an invalid state causing this `nil`error.

**Comment by Gogo1951** [comment]
Nah, it wasn't at sign-in. I think this was in Badlands and we had been questing for a bit... I think a buddy pulled some mobs and I was racing to get the quest for that mob. I think in Badlands, fighting the last stone elemental in the chain. But I forget.

**Comment by BreakBB** [comment]
Yeah, this 100% is an edge case. It not even obviously to me to see how this can possibly happen. Needs more research

**Comment by Gogo1951** [comment]
I got it again today.

Happens when you're questing and you have a mob or are running from a mob (so you're in combat) and you turn in a quest or pick up a new quest.

Basically, the situation is, "I'm not gonna kill that thing, I can sprint / or bubble / or ignore it... so I'm just gonna grab my quest and go."

Then when you snag the quest, while in combat, it gives the Lua error.

**Comment by BreakBB** [comment]
I was not yet able to reproduce this. @Cabro iirc you said something like this happened to you lately?

**Comment by Cabro** [comment]
Not this particular one, but then again i dont usually turn in/accept new quests while in combat. The issue i had was with tracker not wanting to abandon a quest i picked up while in combat

**Comment by Gogo1951** [comment]
I just got this again today. Can confirm it's still an issue, but I don't know how to consistently reproduce it. Happens when I turn quests in while in combat. I like to see how fast I can go through zones, and typical to grab all the quests I can even if I have to run through some mobs to get to the quest givers.
